### ISSUE TITLE
Internal feeds should be disabled by default

### ISSUE THREAD

**Issue by mmitche** [dotnet/arcade#5890]
- [ ] This issue is blocking 
- [x] This issue is causing unreasonable pain

This is part of a larger effort to make public builds of recently open sourced internal fixes work better. Today there are two problems:
- Customers want to build the shas we built, and if those shas were built internally, they have "darc-int" feeds in them. These feeds are not accessible externally. On release day, these feeds aren't necessary. The packages have already been published to public locations (nuget.org). Often by the time a customer goes to build, these feeds don't even have any packages.
- The PRs opened to merge into the open have to be altered to remove the darc-int feeds, or otherwise the builds will fail.

The darc-int (and darc-pub, but those are public) feeds are added because they contain stable packages. To avoid the issues we see today, all darc-int feeds will be disabled **by default** in nuget.config files. Public builds of previously internal sources will work as expected on release day, and would fail prior (because not all packages are available). Today we have to run a script (SetupNuGetSources.ps1/sh) prior to building to add credentials for those feeds. This script will be altered to re-enable these sources if running on the internal AzDO project.

There are 3 parts to this change:
- [x] Arcade **master**, In SetupNuGetSources.ps1/sh, the script should identify the disabledPackageSources node. If it exists, find any darc-int feeds (use the same identification mechanism we use when adding credentials) and flip `value="true"` to `value="false"`. If no darc-int feeds exist there, or the node does not exist, do nothing.
- [x] Arcade **release/3.x**, In SetupNuGetSources.ps1/sh, the script should identify the disabledPackageSources node. If it exists, find any darc-int feeds (use the same identification mechanism we use when adding credentials) and flip `value="true"` to `value="false"`. If no darc-int feeds exist there, or the node does not exist, do nothing.
- [x] Maestro should add new functionality (and tests. there are already tests for manipulation of nuget.config) to add a "disabledPackageSources" node if it does not exist, and its contents should either be empty (but **not** `<clear/>`), or they should contain the source names of all darc-int sources, with the `value="true"` attribute.

```
<disabledPackageSources>
    <add key="darc-int-dotnet-core-setup-0c2e69ca" value="true" />
</disabledPackageSources>
```

**Comment by mmitche** [comment]
As far as rollout goes, the first two checks should be done prior to **rollout** of the Maestro change. The following repos/branches should get updated arcades with the SetupNuGetSources.ps1.

Best effort should be made to ensure that the repos have gotten updated arcades prior to rollout. I would not block on every 3.x and 5 repo getting the update prior to rollout though, as the fixup if it breaks is simple (get the updated SetupNuGetSources.ps1/sh in the branch, or force the arcade update), and I do not want this to drag on (it's a bit of a fire).

**Comment by mmitche** [comment]
Maestro should also add a comment around the feeds in the disabled package sources section.

**Comment by dagood** [comment]
I've filed a source-build issue to react to this change: https://github.com/dotnet/source-build/issues/1696.

**Comment by riarenas** [comment]
https://github.com/dotnet/arcade-services/blob/da13203439b3dc2464e10814892100448209706f/src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs#L327 is where Maestro should add the new disabledPackageSources node. It already takes the set of managed feeds, so it's just a matter of adding the new node, and sticking any `darc-int feeds` in there.

**Comment by riarenas** [comment]
Reference for disabledPackageSources: https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#disabledpackagesources

**Comment by MattGal** [comment]
With https://github.com/dotnet/arcade-services/pull/1334/files merged (checkbox 3), this goes to rollout.

**Comment by MattGal** [comment]
Rolled out yesterday, will work on follow-up part today.
